### PR TITLE
Introduce new command line parameter --track-revision

### PR DESCRIPTION
**Pull request by ebadyano**
Allows a user to run against an older commit of a Rally track.

Fixes https://github.com/elastic/rally/issues/663

### CODE REVIEW

File: esrally/rally.py [hunk 1]
```diff
@@ -167,6 +167,10 @@ def positive_number(v):
         track_source_group.add_argument(
             "--track-path",
             help="Define the path to a track.")
+        track_source_group.add_argument(
+            "--track-revision",
+            help="Define a specific revision in the tracks repository that Rally should use.",
```
Comment (danielmitterdorfer) [inline esrally/rally.py]: nit: `track repository`

File: docs/command_line_reference.rst [hunk 2]
```diff
@@ -64,6 +64,11 @@ Examples::
 
 Selects the track repository that Rally should use to resolve tracks. By default the ``default`` track repository is used, which is available in the Github project `rally-tracks <https://github.com/elastic/rally-tracks>`_. See the :doc:`track reference </track>` on how to add your own track repositories. ``--track-path`` and ``--track-repository`` as well as ``--track`` are mutually exclusive.
 
+``track-revision``
+~~~~~~~~~~~~~~~~~
```
Comment (danielmitterdorfer) [inline docs/command_line_reference.rst]: When building the docs, this produces the following warning:

```

``track-revision``
~~~~~~~~~~~~~~~~~
~/Projects/rally/docs/command_line_reference.rst:68: WARNING: Title underline too short.
```

Can you please fix the warning?

Comment (ebadyano) [reply docs/command_line_reference.rst]: Fixed

File: esrally/rally.py (hunk 1 above)
Comment (ebadyano) [reply esrally/rally.py]: @danielmitterdorfer Thank you for the review. I addressed you comments. Thanks!

**Pull request merged**